src/search/screens/search.screen.js

```diff
@@ -172,26 +209,58 @@ class Search extends Component {
     return <UserListItem user={item} navigation={this.props.navigation} />;
   };
 
+  renderFooter = () => {
+    const { searchType } = this.state;
+
+    if (
+      this.props[
+        searchType === SearchTypes.REPOS
+          ? 'reposSearchResultsPagination'
+          : 'usersSearchResultsPagination'
+      ].nextPageUrl === null
+    ) {
+      return null;
+    }
```

Comment:
Here, we conditionally select the good pagination. 

I can brake the code to have something less barbaric. This will change anyways when we introduce Code & Wikis search

Reply:
Just wondering, how are you thinking we can simplify this when code & wikis are introduced?

Also, in the meantime, maybe we can simplify the following to something like:

```
renderFooter = () => {
    const { searchType } = this.state;
    const { reposSearchResultsPagination, usersSearchResultsPagination } = this.props;

    const currResultsPagination = searchType === SearchTypes.REPOS ? reposSearchResultsPagination : usersSearchResultsPagination;

    if (currResultsPagination.nextPageUrl) {
      return (
        <View
          style={{
            paddingVertical: 20,
          }}
        >
          <ActivityIndicator animating size="large" />
        </View>
      );
    }
  };
```

And another thought (if not know) - maybe we can extract this to a separate component (`<LoadingFooter />`?) if this logic is duplicated exactly the same in multiple screens?

Reply:
I was thinking about having `getActiveResultsPagination()` & `getActiveResults()` instead of using ifs everywhere.

And for the refactoring, I was thinking of:
* a `<PaginatedFlatList>` which would take a `pagination` prop and do everything for us. 
* a `getPagination()` in utils to avoid this chunk of code in mapStateToProps:

```
  const reposSearchResultsPagination = SEARCH_REPOS[searchQuery] || {
    ids: [],
    isFetching: false,
 }
```

was just holding off on changing stuff until I got a few PRs merged. I think I'll do that next